atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/HeartbeatServiceIntegrationTest.java

```diff
@@ -37,7 +38,10 @@
 import com.palantir.atlasdb.containers.CassandraContainer;
 import com.palantir.atlasdb.containers.Containers;
 import com.palantir.atlasdb.keyvalue.impl.TracingPrefsConfig;
+import com.palantir.flake.FlakeRetryingRule;
+import com.palantir.flake.ShouldRetry;
 
+@ShouldRetry // There are flakes with the heartbeat service throwing because it was unable to stop beating.
```

Comment:
Adding comments to explain why the test is flaking seems like good practice.  Maybe we should enforce in the future by adding a required `reason` field to the annotation?

Reply:
+1. This would be good to do before broader rollout of the flake-rule as well (so we don't have people shouting about API breaks!)

Reply:
also, yes - retry on specific exception type is planned for the next part.